### PR TITLE
feat(ihev2): ehex + fixing dr chunking

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1667](https://github.com/metriport/metriport-internal/issues/1667)

### Description

- break up DQs for ehex by time - [context](https://metriport.slack.com/archives/C065BLRBUDQ/p1720010852229919)
- fixed dr chunking to look at prefixes for certain oids instead of exact match
- fixing dr errors in prod by making certain schema values optional

### Testing

- Local
  - [x] unit tests
- Staging
  - [x] branch to staging
  - [ ] staging

Check each PR.

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/carequality/ihe-gateway-v2/outbound/__tests__/create-xcpd.test.ts
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/outbound/__tests__/create-xcpd.test.ts:1]: this test is redundant

File: packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/schema.ts
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/schema.ts:1]: making certain schema values optional to fix dr errors in prod

File: packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts [hunk 1]
```diff
@@ -130,8 +130,12 @@ async function processDocumentReference({
...
       metriportId: metriportId,
       fileLocation: bucket,
-      homeCommunityId: stripUrnPrefix(documentResponse.HomeCommunityId),
-      repositoryUniqueId: documentResponse.RepositoryUniqueId,
+      homeCommunityId: documentResponse.HomeCommunityId
+        ? stripUrnPrefix(documentResponse.HomeCommunityId)
+        : outboundRequest.gateway.homeCommunityId,
+      repositoryUniqueId: documentResponse.RepositoryUniqueId
+        ? stripUrnPrefix(documentResponse.RepositoryUniqueId)
+        : outboundRequest.gateway.homeCommunityId,
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts:138]: using request home community id in the case where we didnt get one

File: packages/api/src/external/carequality/document/create-outbound-document-query-req.ts [hunk 2]
```diff
@@ -21,34 +22,78 @@ export function createOutboundDocumentQueryRequests({
...
-      id: requestId,
-      cxId: cxId,
-      timestamp: now,
-      samlAttributes: {
-        subjectId: user,
-        subjectRole: {
-          code: SUBJECT_ROLE_CODE,
-          display: SUBJECT_ROLE_DISPLAY,
+  return cqLinks.flatMap(externalGateway => {
+    if (doesGatewayNeedDateRanges(externalGateway.url)) {
```
Comment (leite08) [inline packages/api/src/external/carequality/document/create-outbound-document-query-req.ts:29]: There's a lot of logic here, can we move this to a function that we call in the `.flatMap()`, please?

Comment (leite08) [reply packages/api/src/external/carequality/document/create-outbound-document-query-req.ts:29]: Also, both "sides" of the `if` build the request pretty much the same, just the first one adds `documentCreationDate`. Maybe we create them first, and then check if the requests need to be further broken down and updated?

File: packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts (hunk 1 above)
Comment (leite08) [reply packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts:138]: Just to make sure it's not a copy/paste typo, for `repositoryUniqueId` we also fallback to the request's `homeCommunityId`, right?

File: packages/api/src/external/carequality/document/create-outbound-document-query-req.ts [hunk 3]
```diff
@@ -9,6 +9,105 @@ import { doesGatewayNeedDateRanges } from "@metriport/core/external/carequality/
...
+  patient: Patient;
+}): OutboundDocumentQueryReq[] {
+  const requests = [];
+  const length = 10;
+  for (let i = 0; i < length; i++) {
+    const dateTo = now.subtract(i * 6, "month").toISOString();
+    const dateFrom =
+      i !== length - 1
+        ? now.subtract((i + 1) * 6, "month").toISOString()
+        : now.subtract(25, "year").toISOString();
```
Comment (leite08) [inline packages/api/src/external/carequality/document/create-outbound-document-query-req.ts:93]: 25 years seem a pretty important threshold to be "hidden" in the middle of this code. Can we move it to a constant on the top of this file, please?

**Review by leite08 — APPROVED** [review]
Approving conditional to exposing the threshold as a global constant. 🚀

**Pull request merged**